astropy/utils/xml/writer.py

```diff
@@ -148,6 +153,20 @@ def start(self, tag, attrib={}, **extra):
         return len(self._tags)
 
     @contextlib.contextmanager
+    def xml_escaping(self, enabled=True, clean_kwargs=None):
+        current_xml_escape_cdata = self.xml_escape_cdata
+        if not enabled:
+            if HAS_BLEACH:
+                if clean_kwargs is None:
+                    clean_kwargs = {}
+                self.xml_escape_cdata = lambda x: bleach.clean(x, **clean_kwargs)
+            else:
+                raise ValueError('bleach package is required when HTML escaping is disabled.\n'
+                                 'Use "pip install bleach".')
```

Comment:
That strategy forces any user of `xml_escaping` to essentially pre-check whether there will be ever be raw HTML.  That might be inconvenient if the code doesn't have available prior knowledge of this information in order to raise an exception at a convenient point.  I *think* this strategy leads to code like:
```
# Pre-check (somehow...)
user_wants_raw_html = check_if_user_wants_raw_html_anywhere(...)

if user_wants_raw_html:
    try:
        import bleach
    except ImportError:
        raise ImportError('you need bleach if you want raw HTML')

# Later on, possibly many layers deep in the code
user_wants_raw_html_here = ...

if user_wants_raw_html_here:
    # Allow raw HTML (disable brute force HTML escaping) but use bleach.clean()
    # to sanitize dangerous tags
    with w.xml_escaping(False, raw_html_clean_kwargs):
        w.start('td')
        w.data(some_data)
        w.end(indent=False)
else:
    # Use brute-force escaping, but user might not have bleach installed, 
    # cannot use w.xml_escaping(True, ..)
    w.start('td')
    w.data(el.strip())
    w.end(indent=False)
```